tests/test_multithreaded_iterator.py

```diff
@@ -0,0 +1,107 @@
+import subprocess as sb
+
+
+def run(command):
+    return sb.run(command.split(), capture_output=True) \
+               .stdout \
+               .decode() \
+               .strip()
+
+def hash_password(password, salt="hi"):
+    return run(f"./encr -p {password} -s {salt}")
+
+# Length 1
+def test_st_iter_length1_first():
```

Comment:
Too many copy/paste. Repeated code should be factored out to a function. Majority of the tests will be a call of this function